### PR TITLE
Relocate reusable integration tests components to core

### PR DESCRIPTION
**Pull request by paweljakubas**
# Issue Number

#358 

# Overview

- [x] I have relocated integration test components from http-bridge to core and divided Faucet

# Comments

### CODE REVIEW

File: stack.yaml [hunk 1]
```diff
@@ -3,6 +3,7 @@ packages:
 - .
 - lib/bech32
 - lib/core
+- lib/core/test/integration
```
Comment (Anviking) [inline stack.yaml]: I guess this makes sense, but I'm curious if you considered not having `core` in the path/name: e.g `lib/integration`?

File: lib/http-bridge/cardano-wallet-http-bridge.cabal [hunk 2]
```diff
@@ -65,6 +65,7 @@ library
       Cardano.Wallet.HttpBridge.Transaction
       Data.Packfile
       Servant.Extra.ContentTypes
+      Paths_cardano_wallet_http_bridge
```
Comment (Anviking) [inline lib/http-bridge/cardano-wallet-http-bridge.cabal]: Unrelated, but had no idea [this](https://stackoverflow.com/questions/33176905/is-there-a-way-to-get-version-of-cabal-package-in-source-code-of-the-package/33176906) was possible 😮

**Review by Anviking — COMMENTED** [review]
lgtm!

**Comment by Anviking** [comment]
I rebased and removed unused `LambdaCase` in `http-bridge/test/integration/Cardano/Faucet.hs` to satisfy failing hlint.

File: lib/core/test/integration/cardano-wallet-core-integration.cabal [hunk 3]
```diff
@@ -0,0 +1,68 @@
+name:                cardano-wallet-core-integration
```
Comment (KtorZ) [inline lib/core/test/integration/cardano-wallet-core-integration.cabal]: Let's not make it a library, but have it simply located in `core/test` and referred to in the corresponding integration tests using `hs-source-dirs`. Having test code as a library will have further complications.

Comment (paweljakubas) [reply lib/core/test/integration/cardano-wallet-core-integration.cabal]: good idea, addressed in new commit

**Comment by paweljakubas** [comment]
I have weeder complaining : 
```
== Section test:integration ==
Missing other-modules entry
* Test.Integration.Faucet
* Test.Integration.Framework.DSL
* Test.Integration.Framework.Request
* Test.Integration.Scenario.API.Addresses
* Test.Integration.Scenario.API.Transactions
* Test.Integration.Scenario.API.Wallets
* Test.Integration.Scenario.CLI.Addresses
* Test.Integration.Scenario.CLI.Mnemonics
* Test.Integration.Scenario.CLI.Transactions
* Test.Integration.Scenario.CLI.Wallets
Module not compiled
* Test.Integration.Faucet
* Test.Integration.Framework.DSL
* Test.Integration.Framework.Request
* Test.Integration.Framework.TestData
* Test.Integration.Scenario.API.Addresses
* Test.Integration.Scenario.API.Transactions
* Test.Integration.Scenario.API.Wallets
* Test.Integration.Scenario.CLI.Addresses
* Test.Integration.Scenario.CLI.Mnemonics
* Test.Integration.Scenario.CLI.Transactions
* Test.Integration.Scenario.CLI.Wallets
Redundant build-depends entry
* aeson-qq
* command
* cryptonite
* exceptions
* generic-lens
* hspec-expectations-lifted
* http-api-data
* http-types
* template-haskell
```

**Review by KtorZ — APPROVED** [review]
:+1:

**Comment by paweljakubas** [comment]
I wonder what to do with weeder - refactor cabal or add appropriate section to .weeder.yaml?

**Pull request merged**